examples/azure.ts

```diff
@@ -1,14 +1,19 @@
 #!/usr/bin/env -S npm run tsn -T
 
 import { AzureOpenAI } from 'openai';
+import { getBearerTokenProvider, DefaultAzureCredential } from '@azure/identity';
 
 // Corresponds to your Model deployment within your OpenAI resource, e.g. gpt-4-1106-preview
 // Navigate to the Azure OpenAI Studio to deploy a model.
 const deployment = 'gpt-4-1106-preview';
 
+const credential = new DefaultAzureCredential();
+const scope = 'https://cognitiveservices.azure.com/.default';
+const azureADTokenProvider = getBearerTokenProvider(credential, scope);
+
 // Make sure to set both AZURE_OPENAI_ENDPOINT with the endpoint of your Azure resource and AZURE_OPENAI_API_KEY with the API key.
 // You can find both information in the Azure Portal.
```

Comment:
nit: this comment is outdated now right? I think we should list out the different ways to authenticate here

Reply:
Good catch, addressed in https://github.com/openai/openai-node/pull/857/commits/32ee4eb3f40aece122713d68c29fb2e5cce2b970.